### PR TITLE
Add graphql permissions

### PR DESCRIPTION
**Pull request by ZachCMP**
https://www.pivotaltracker.com/story/show/184497361

Requires backend code in https://github.com/greenriver/hmis-warehouse/pull/2840

### CODE REVIEW

File: src/modules/permissions/useHasPermissionsHooks.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+export type ProjectPermissions = keyof Omit<ProjectAccess, '__typename'>;
+export const useHasProjectPermissions = (
+  id: string,
+  permissions: ProjectPermissions[],
+  mode?: PermissionsMode
+) => {
+  const projectPermissions = useGetProjectPermissionsQuery({
+    variables: { id },
+  })?.data?.project?.access;
+  return useHasPermissions(projectPermissions, permissions, mode);
```
Comment (gigxz) [inline src/modules/permissions/useHasPermissionsHooks.tsx]: We shouldn't ignore the loading state here

File: src/components/pages/EditProject.tsx [hunk 2]
```diff
@@ -44,6 +49,7 @@ const EditProject = () => {
 
   if (loading) return <Loading />;
```
Comment (gigxz) [inline src/components/pages/EditProject.tsx]: This shoudl consider whether the project permission query is loading too

File: src/components/pages/Organization.tsx [hunk 3]
```diff
@@ -72,6 +72,7 @@ const Organization = () => {
           )}
         </Grid>
         <Grid item xs>
+          {organization?.contactInformation}
```
Comment (gigxz) [inline src/components/pages/Organization.tsx]: Why is this added?

File: src/components/pages/EditProject.tsx (hunk 2 above)
Comment (gigxz) [reply src/components/pages/EditProject.tsx]: Actually, this is wrapped in `ProjectEditRoute`. So we shouldn't need to make. project permission query in this component at all.

**Review by gigxz — CHANGES_REQUESTED** [review]

File: src/components/pages/Organization.tsx (hunk 3 above)
Comment (ZachCMP) [reply src/components/pages/Organization.tsx]: I had made the create project action use the `can_edit_project_details` permission, in which case we should totally hide this paper if there's no contact info and we can't create a project. So I think this was initially a condition to check for that. I thought better of using that permission, so I removed the permission filter, but must have not completely removed the code that would hide the paper, thus this line. I'll remove it

File: src/components/pages/EditProject.tsx (hunk 2 above)
Comment (ZachCMP) [reply src/components/pages/EditProject.tsx]: fixed

File: src/modules/permissions/useHasPermissionsHooks.tsx (hunk 1 above)
Comment (ZachCMP) [reply src/modules/permissions/useHasPermissionsHooks.tsx]: I made this return a tuple with the form `[<permission boolean>, <query data>]` so that we can get everything we could need from query data.

**Comment by ZachCMP** [comment]
@gigxz Addressed all your feedback. I'll branch off this for the rest of the react work around the root permissions

**Review by gigxz — APPROVED** [review]
